File: packages/sdk-python/steno_ai/sessions.py
"""Session client — start and end memory sessions."""

from .client import HttpClient


class SessionClient:
    def __init__(self, http: HttpClient):
        self._http = http

    def start(self, scope: str, scope_id: str) -> dict:
        """Start a new memory session."""
        return self._http.request("POST", "/v1/sessions", json={
            "scope": scope,
            "scope_id": scope_id,
        })

    def end(self, session_id: str) -> None:
        """End an active session."""
        self._http.request("POST", f"/v1/sessions/{session_id}/end")

File: packages/sdk-ts/src/clients/sessions.ts
import type { HttpClient } from '../http.js';
import type { Session } from '../types.js';

export class SessionClient {
  constructor(private readonly http: HttpClient) {}

  /** Start a new memory session. */
  async start(scope: string, scopeId: string): Promise<Session> {
    return this.http.request('POST', '/v1/sessions', { scope, scopeId });
  }

  /** End an active session. */
  async end(sessionId: string): Promise<void> {
    return this.http.request('POST', `/v1/sessions/${sessionId}/end`);
  }
}
